src/main/java/com/ace/alfox/controllers/ActionController.java

```diff
@@ -1,35 +1,35 @@
-package com.ace.alfox;
-
-import com.ace.alfox.game.GameCharacter;
-import com.ace.alfox.lib.Action;
-import com.ace.alfox.lib.CharacterFactory;
-import com.ace.alfox.lib.Result;
-import org.springframework.http.*;
-import org.springframework.web.bind.annotation.*;
-
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpSession;
-
-import java.io.*;
-import java.util.Map;
-
-@RestController
-class ActionController {
-
-    @RequestMapping(value = "/perform/{command}", method = { RequestMethod.GET, RequestMethod.POST })
-    public ResponseEntity home(HttpServletRequest request, @PathVariable String command, @RequestBody Map<String, Object> params) throws IOException {
-        HttpSession session = request.getSession(true);
-        String characterId = (String) session.getAttribute("cid");
-
-        GameCharacter player = CharacterFactory.fetch(characterId);
-        if(player == null) {
-            return ResponseEntity.notFound().build();
-        }
-
-        Action doSomething = Action.named(command).withParameters(params);
-        Result result = player.applyAction(doSomething);
-        CharacterFactory.save(result.character);
-
-        return ResponseEntity.ok(result);
-    }
+package com.ace.alfox.controllers;
+
+import com.ace.alfox.game.models.Player;
+import com.ace.alfox.lib.ActionFactory;
+import com.ace.alfox.lib.ActionWebResult;
+import com.ace.alfox.lib.CharacterFactory;
+import org.springframework.http.*;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+import java.io.*;
+import java.util.Map;
+
+@RestController
+class ActionController {
+
+    @RequestMapping(value = "/perform/{command}", method = { RequestMethod.GET, RequestMethod.POST })
+    public ResponseEntity home(HttpServletRequest request, @PathVariable String command, @RequestBody Map<String, Object> params) throws IOException {
+        HttpSession session = request.getSession(true);
+        String characterId = (String) session.getAttribute("cid");
+
+        Player player = CharacterFactory.fetch(characterId);
```

Comment:
Might be good to align character/player. ie instead of a characterId being used to create a Player, we have a playerId. Or vice versa.

Reply:
Yup, totally missed this. I originally used `Character`, but that's a type in `java.lang` I think and I wanted to pick something different.